documentation/projects/proposals/monitoring/20230606_implementation_plan_ecs_alarms.md

```diff
@@ -0,0 +1,595 @@
+stabilised, with relevant run books updated to reflect that. To summarise the
+overall process:
+
+1. Propose a new alarm based on metric observation
+2. Implement the unstable version of the alarm and create the initial run book
+   and link it from the notification for the alarm
+3. Evaluate the new alarm over time to ensure correctness
+4. Iterate on the severity identification guide for alerts from the alarm based
+   on experience evaluating the unstable alarm
+5. Stabilise the alarm by updating the run book and sending alerts to the stable
+   channel
+
+### Run Books
+
+Alert run books document the process for understanding a particular alert from
+one or many alarms. Alert run books are not replacements for the general
+incident response plan and maintainers must defer to the general process for
+incidence response. The primary responsibility of the run book is to help
+maintainers identify the severity of an issue and identify false positives.
+While an alarm is still unstable, the run book also serves to document
+historical information relevant to false positive identification and potential
+changes to the alarm configuration that may help the alarm reach stability.
+Alert run books should also document downtime.
+
+Every notification sent by an alarm must include a link to a run book.
+
+Run books are critical to alarm development because they document and share the
+knowledge of metric interpretation with everyone responding to alerts. Some
+alarms may share similar or exactly the same details or instructions. In those
+cases, we can develop documents that are linked to from individual alert run
+books. Even if every detail is shared between two separate alarms, each alarm
+must still have its own run book to serve as a space to encourage alarm-specific
+documentation. An alarm will naturally have few details in its run book to begin
+with, but it should expand over time as maintainers' experience with each alarm
+grows.
+
+To clarify: alert run books must help identify severity and false positives and
+may document remediation avenues but should never supersede the general
+incidence response plan.
+
+```{tip}
+Run books are for _alerts_, not alarms. Alerts inform us of relevant alarm
+states. Some alarms may go off for multiple states. For example, anomaly detection may use the same alarm to notify when a metric is too high or too low. Each is a separate "alert" and may require separate information to identify severity. Additional consideration may be necessary for alerts from composite alarms. These may require manually checking multiple alarms to help identify severity. Keeping this in mind helps us focus each run book as much as possible so that responders have the most accessible information at any time.
+```
+
+Run books must include the following:
+
+- Stability: whether the alert is considered stable or unstable
+- The run book maintainer's GitHub handle, if the alert is unstable
+- Configured downtime
```

Comment:
You mention this in a few places, but I'm a little unclear what's meant by it - do you mind adding it to the "Terms" section with an explanation?

Reply:
Good call to add it to terms. It is clarified later in the implementation details section: https://github.com//WordPress/openverse/blob/8ac694b080da7c4b1a36960762c2db6a8743bbbc/documentation/projects/proposals/monitoring/20230606_implementation_plan_ecs_alarms.md#L415-L420